API.md

```diff
@@ -2871,6 +2871,28 @@ const handler = function (request, reply) {
 };
 ```
 
+If the handler returns a Promise then Hapi will register a `catch` handler on the promise object to catch unhandled promise rejections. The handler will `reply` with the error:
+
+```js
+const handler = function (request, reply) {
+
+  const badPromise = () => {
+
+    new Promise(() => {
+
+      setTimeout(() => throw new Error(), 1000);
```

Comment:
Added a test case showing that the `setTimeout` example is already handled by the `protect` mechanism, to keep things clearer. Also attempted to make the api documentation sharper.